src/config/reader.rs

```diff
@@ -62,6 +65,77 @@ impl ConfigReader {
         Ok(content)
     }
 
+    /// Reads the links in a Config and fill the content
+    #[async_recursion::async_recursion]
+    async fn read_links(&self, mut config_set: ConfigSet) -> anyhow::Result<ConfigSet> {
+        let links = config_set.config.links.clone();
+
+        if links.is_empty() {
+            return Ok(config_set);
+        }
+
+        if links.iter().any(|link| link.src.is_empty()) {
+            return Err(anyhow::anyhow!("Link src cannot be empty"));
+        }
+
+        for config_link in links.iter() {
+            let source = self.read_file(&config_link.src).await?;
+            let content = source.content;
+
+            match config_link.type_of {
+                LinkType::Config => {
+                    let config =
+                        Config::from_source(Source::detect(&source.path).unwrap(), &content)?;
+
+                    config_set = config_set.merge_right(&ConfigSet::from(config.clone()));
+
+                    if !config.links.is_empty() {
+                        config_set = config_set
+                            .merge_right(&self.read_links(ConfigSet::from(config)).await?);
+                    }
+                }
+                LinkType::GraphQL => {
+                    let service_document =
+                        Config::from_source(Source::detect(&source.path).unwrap(), &content)?
+                            .to_document();
+
+                    config_set
+                        .extensions
+                        .service_document_from_links
+                        .push(ServiceDocumentWithId {
+                            id: config_link.id.to_owned(),
+                            service_document,
+                        });
+                }
+                LinkType::Protobuf => {
```

Comment:
With this feature, we should drop the `protoPath` that's available on the GRPC operator.